### ISSUE TITLE
Xamarin.iOS app with EFCore 2.0 SQLitePCL.Batteries.Init()

### ISSUE THREAD

**Issue by bricelam** [aspnet/EntityFramework.Docs#597]
_From @alexdrl on November 21, 2017 8:44_

A Xamarin.iOS app with EFCore 2.0 needs a call to 

```c#
SQLitePCL.Batteries_V2.Init();
```

If the call is not made, the app initialization throws this error message. 

```
You need to call SQLitePCL.raw.SetProvider(). If you are using a bundle package, this is done by calling SQLitePCL.Batteries.Init().
```
Like explained [here](https://github.com/ericsink/SQLitePCL.raw/wiki/SQLitePCL.Batteries.Init)

### Steps to reproduce
Xamarin.iOS project working with EFCore 2.0. Solution with a common .NET Standard project which shares code between iOS and Android, using PackageReference.

### Further technical details
EF Core version: 2.0.0
Database Provider: Microsoft.EntityFrameworkCore.Sqlite
Operating system: iOS
IDE: Visual Studio 15.4

_Copied from original issue: aspnet/EntityFrameworkCore#10361_

**Comment by bricelam** [comment]
Correct. I don't think Xamarin.iOS can handle [the Assembly.Load() call](https://github.com/aspnet/Microsoft.Data.Sqlite/blob/rel/2.0.0/src/Microsoft.Data.Sqlite.Core/Utilities/BundleInitializer.cs#L15). And binding directly to `SQLitePCLRaw.batteries_v2.dll` will cause problems if you're not using a bundle.

**Comment by bricelam** [comment]
_From @alexdrl on November 22, 2017 18:27_

I use the SQLitePCLraw assembly that gets added as a transitive dependency, and call that Init method, the app works correctly. In the project assets json, the Xamarin.iOS SQLitePCLraw package gets restored correctly.

**Comment by bricelam** [comment]
We should probably document this on the [Supported .NET Implementations](https://docs.microsoft.com/ef/core/platforms/) page or thereabouts.

**Comment by bricelam** [comment]
_From @alexdrl on November 22, 2017 18:31_

The documentation already stated that there are problems in Xamarin.iOS but there is no page that describes this (or other) workarounds.

**Comment by dfoulk** [comment]
@cwrea Thank you for mentioning this. I would've spent hours debugging that issue if I hadn't seen your comment first! Literally saved us hundreds of dollars!

**Comment by cwrea** [comment]
@dfoulk You're welcome. FWIW, since I posted that, there's hope that Visual Studio 15.8 Preview 3 may address these issues. See my comment from a couple of days ago at https://github.com/xamarin/xamarin-macios/issues/4168#issuecomment-398521666 ... I haven't tried the preview myself, yet. Tied up in other project work at the moment.

**Comment by Pinox** [comment]
@cwrea  very weird.  I'm running  ef core 2.1 on my android and native UWP and it works and builds fine (debug and release) , but I'm using Microsoft.Extentions.primitives version 2.0

**Comment by cwrea** [comment]
@Pinox Do you have linking enabled? The oustanding Android issue is specific to having linking enabled.

**Comment by Pinox** [comment]
@cwrea  In release mode I have "SDK Assemblies" and in debug mode "none".
